File: Container with Most Water.py
#Time Complexity: O(n)
#Space Complexity: O(1)

class Solution:
    def maxArea(self, height: List[int]) -> int:
        Max=0;
        low=0;
        high=len(height)-1;
        while low< high:
            area=(high-low)*min(height[low],height[high]);
            Max=max(Max,area);
            if height[low]<height[high]:
                low+=1;
            else:
                high-=1;
        return Max;

File: ContainerWithMostWater.java
//Time Complexity - O(n) as we iterate using two pointers
//Space Complexity - O(1) no extra space other than variables

class Solution {
    public int maxArea(int[] height) {
        int maxAmount=0;
        int left=0;
        int right=height.length-1;

        while(left<right)
        {
            int waterHeight= Math.min(height[left],height[right])*(right-left);
            maxAmount=Math.max(maxAmount,waterHeight);
            if(height[left]<height[right])
                left++;
            else
                right--;
        }
        return maxAmount;
    }
}